Wikipedia/Images/WMFImageController.swift

```diff
@@ -124,14 +123,54 @@ public class WMFImageController : NSObject {
         .then(mainImageBlock)
     }
 
-    /// MARK: - Simple Fetching
+    /**
+    Use the shared session to download an image directly to the receiver's disk cache.
+
+    - parameter url: The URL to download the image from.
+
+    - returns: A promise which is resolved after the data has been successfully written to disk.
+    */
+    public func downloadImageDataInBackground(url: NSURL) -> Promise<Void> {
```

Comment:
IMHO, the amount of work we'd have to do (or have done) to customize SDWebImage is approaching (if not past) the point of diminishing returns. The author even mentions that [you probably don't need it](https://github.com/rs/SDWebImage/wiki/How-is-SDWebImage-better-than-X%3F#since-afnetworking-provides-similar-functionality-for-uiimageview-is-sdwebimage-still-useful) if you're using a library like `AFNetworking` which provides in-memory caching.  That, coupled with the fact that we're using _very little_ of `SDWebImage` disk cache logic (i.e. not letting it cleanup, customizing the location), I'm starting to think we should contemplate using plain old `NSURLSession` (either directly or through something else) and dropping SDWebImage.

I'm also considering this for our networking stack, but I've yet to closely inspect what AFNetworking features we're actually benefiting from compared to the cost of add'l abstraction.

Ranting aside, this approach is:
- Only in the 10's of lines of code
- Optimized to avoid unnecessary image inflation & memory costs
- Debatably easier to move to using a background `NSURLSession`, which could let us reap add'l benefits for users (by making these & other tasks low-priority, discretionary tasks that can continue in the background **and launch the app to trigger background fetches**)